e2e/testcases/apiservice_test.go

```diff
@@ -52,36 +53,62 @@ func TestCreateAPIServiceAndEndpointInTheSameCommit(t *testing.T) {
 		nt.T.Fatalf("validate failed %s", err)
 	}
 
+	// Test done, removing the test APIService first to prevent Discovery failure blocking
+	// the test repo from cleaning up
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/cluster/apiservice.yaml")
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter API service")
 	nt.WaitForRepoSyncs()
 
+	// Remove the backend Deployment of test APIService
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/custom-metrics/namespace-custom-metrics.yaml")
-	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter deployment")
-	nt.WaitForRepoSyncs()
-
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/custom-metrics/namespace.yaml")
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter namespace")
 	nt.WaitForRepoSyncs()
 }
 
 func TestImporterAndSyncerResilientToFlakyAPIService(t *testing.T) {
 	nt := nomostest.New(t, ntopts.RequireGKE(t), ntopts.Unstructured)
-	nt.T.Log("Adding API service with backend temporarily unavailable")
-	nt.MustKubectl("apply", "-f", "../testdata/apiservice/apiservice.yaml")
 	nt.T.Cleanup(func() {
 		nt.MustKubectl("delete", "-f", "../testdata/apiservice/apiservice.yaml", "--ignore-not-found")
+		nt.MustKubectl("delete", "-f", "../testdata/apiservice/namespace-custom-metrics.yaml", "--ignore-not-found")
+		nt.MustKubectl("delete", "-f", "../testdata/apiservice/namespace.yaml", "--ignore-not-found")
+		nt.MustKubectl("delete", "-f", "../testdata/apiservice/rbac.yaml", "--ignore-not-found")
+
 		if t.Failed() {
 			nt.PodLogs(adapterNamespace, adapterName, "pod-custom-metrics-stackdriver-adapter", true)
 		}
 	})
 
-	nt.T.Log("Creating commit with resources")
-	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/rbac.yaml", "acme/cluster/rbac.yaml")
+	nt.T.Log("Adding APIService with backend temporarily unavailable to trigger discovery errors")
+	nt.MustKubectl("apply", "-f", "../testdata/apiservice/apiservice.yaml")
+
+	nt.T.Log("Creating commit with test resources")
+	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/namespace-resilient.yaml", "acme/namespaces/resilient/namespace.yaml")
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("add testing resources")
 
-	nt.T.Log("Adding backend")
+	if nt.MultiRepo {
+		nt.T.Log("Wait for test resource to have status CURRENT")
+		nt.WaitForRepoSyncs()
+		_, e := nomostest.Retry(nt.DefaultWaitTimeout, func() error {
+			var err error
+			err = nt.Validate("resilient", "", &corev1.Namespace{},
+				nomostest.StatusEquals(nt, kstatus.CurrentStatus))
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+		if e != nil {
+			nt.T.Fatal("validate failed test resource to have status CURRENT")
+		}
```

Comment:
Just to confirm, if the APIService is unavailable in the multi-repo mode, there won't be any effect to the sync and reconciliation status. Basically, user won't notice any difference. Is that the expected behavior?

Reply:
In this test scenario where APIService is unavailable what you described is correct. I was also working on producing a case where multi-repo mode fails to reconcile due to timeout discovery failure until this bats test starts failing. Will switch back to that part once CI is fixed.

Reply:
SG